pkg/controllers/jivavolume_controller.go

```diff
@@ -179,6 +191,30 @@ func (r *JivaVolumeReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return reconcile.Result{}, nil
 }
 
+func (r *JivaVolumeReconciler) updatePodIPMap(cr *openebsiov1alpha1.JivaVolume) error {
+	var (
+		controllerLabel = "openebs.io/component=jiva-controller,openebs.io/persistent-volume="
+	)
+
+	labelSelector, _ := labels.Parse(
+		controllerLabel + cr.Name)
+
+	pods := corev1.PodList{}
+	err := r.List(context.TODO(), &pods, &client.ListOptions{
+		Namespace:     cr.Namespace,
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		return err
+	}
+	if len(pods.Items) != 1 {
+		return fmt.Errorf("expected 1 controller pod got %d", len(pods.Items))
+	}
```

Comment:
What if pods remains in evicted state due to draining of node?

Reply:
Added check with fieldSelector